PlayCover/Views/MainView.swift

```diff
@@ -226,3 +302,98 @@ struct MainView_Previews: PreviewProvider {
             .environmentObject(AppIntegrity())
     }
 }
+
+struct AddFolderView: View {
+    @State var newFolder = ""
+    @Binding var addFolderSheet: Bool
+    var body: some View {
+        VStack {
+            TextField(text: $newFolder, label: {Text("preferences.textfield.url")})
+            Spacer()
+                .frame(height: 20)
+            HStack {
+                Spacer()
+                Button {
+                    addFolderSheet.toggle()
+                } label: {
+                    Text("button.Cancel")
+                }
+                Button {
+                    addFolderSheet.toggle()
+                } label: {
+                    Text("button.OK")
+                }
+                .tint(.accentColor)
+                .keyboardShortcut(.defaultAction)
+            }
+        }
+        .padding()
+        .frame(width: 600, height: 100)
+    }
+}
+
+struct Folder: Identifiable, Codable {
+    let id: UUID
+    var name: String
+    var apps: [String]
+    var icon: String = "folder"
+    init(name: String, icon: String) {
+        self.id = UUID()
+        self.name = name
+        self.apps = []
+        self.icon = icon
+    }
+}
+
+class AppFolder: ObservableObject {
+    static let shared = AppFolder()
+
+    @Published var folders: [Folder] {
+        didSet {
+            encode()
+        }
+    }
+
+    var plistFolderApps = PlayTools.playCoverContainer
+        .appendingPathComponent("appFolders")
+        .appendingPathExtension("plist")
+
+    func addFolder(folder: String, icon: String) {
+        self.folders.append(Folder(name: folder, icon: icon))
+        }
+
+    func encode() {
+        let encoder = PropertyListEncoder()
+        encoder.outputFormat = .xml // .xml is usually preferred for .plist
+
+        do {
+            let data = try encoder.encode(self.folders)
+            try data.write(to: plistFolderApps)
+            print("Folders saved to: \(plistFolderApps)")
+        } catch {
+            print("Error saving folders to .plist: \(error)")
+        }
+    }
+
+    init() {
```

Comment:
The `init` should be right below variable initialization and above function initialization.